src/OpenTelemetry/Metrics/MeterProviderBuilderSdk.cs

```diff
@@ -79,6 +81,93 @@ internal MeterProviderBuilderSdk AddMetricReader(MetricReader metricReader)
             return this;
         }
 
+        internal MeterProviderBuilderSdk AddViewCallback(Func<Instrument, MetricStreamConfig> callback)
+        {
+            this.viewConfigs.Add(callback);
+            return this;
+        }
+
+        internal MeterProviderBuilderSdk AddView(string name, string meterName, string meterVersion, string instrumentName, InstrumentType instrumentType, string[] tagKeys, Aggregation aggregation, double[] histogramBounds)
+        {
+            if (string.IsNullOrWhiteSpace(meterVersion)
+                && string.IsNullOrWhiteSpace(meterName)
+                && string.IsNullOrWhiteSpace(instrumentName))
+            {
+                throw new ArgumentException("Atleast one instrument selection criteria should be provided.");
+            }
+
+            if (histogramBounds != null && aggregation != Aggregation.Histogram)
+            {
+                throw new ArgumentException("Histogram bounds are only applicable if Aggregation is Histogram.");
+            }
+
+            Func<Instrument, MetricStreamConfig> viewConfig = (instrument) =>
+            {
+                bool selectInstrument = false;
+                if (!string.IsNullOrWhiteSpace(meterName)
+                && instrument.Meter.Name.StartsWith(meterName, StringComparison.OrdinalIgnoreCase))
+                {
+                    selectInstrument &= true;
+                }
+
+                if (!string.IsNullOrWhiteSpace(meterVersion)
```

Comment:
Since these 3 string are already checked for `IsNullOrWhiteSpace` should we cache that value? Or can we indicate to the compiler this string will never change, make them `in` arguments? https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/in-parameter-modifier